MakeHist/mumu.C

```diff
@@ -0,0 +1,337 @@
+                               }
+                               if(Jet2_nonB_PT < Jets[Nj].Pt() && Jet1_nonB_PT > Jets[Nj].Pt()){
+                                   Jet2_nonB = Nj;
+                                   Jet2_nonB_PT=Jets[Nj].Pt();
+                               }
+                           }
+                       }
+
+                       else if(Nbjet == 1){
+                           for(unsigned Nj=0; Nj<Jets.size(); ++Nj){
+                               if(Jet2_nonB_PT<Jets[Nj].Pt()){
+                                   if(Jet2_nonB_PT != bJets[0].Pt()){
+                                       Jet2_nonB=Nj;
+                                       Jet2_nonB_PT=jet_pt[Nj];
+                                   }
+                               }
+                           }
+                       }
+                       else if(Nbjet > 1){
+                           for(unsigned Nbj=0; Nbj<bJets.size(); ++Nbj){
+                               All_bjet_PT += bJets[Nbj].Pt();
+                               if(Jet1_B_PT < bJets[Nbj].Pt()){
+                                   Jet1_B=Nbj;
+                                   Jet1_B_PT=bJets[Nbj].Pt();
+                               }
+                               if(Jet2_B_PT < bJets[Nbj].Pt() && Jet1_B_PT > bJets[Nbj].Pt()){
+                                   Jet2_B = Nbj;
+                                   Jet2_B_PT = bJets[Nbj].Pt();
+                               }
+                           }
+                       }
+
+                       TLorentzVector met_p4;
+
+                       float lep1_jet1_mass;
+                       float lep1_jet2_mass;
+                       float lep2_jet1_mass;
+                       float lep2_jet2_mass;
+
+                       if(Nbjet == 0){
+                           lep1_jet1_mass = ((leptons[1])+(Jets[Jet1_nonB])).M();
+                           lep1_jet2_mass = ((leptons[1])+(Jets[Jet2_nonB])).M();
+                           lep2_jet1_mass = ((leptons[2])+(Jets[Jet1_nonB])).M();
+                           lep2_jet2_mass = ((leptons[2])+(Jets[Jet2_nonB])).M();
+                       }
+
+                       else if(Nbjet == 1){
+                           lep1_jet1_mass = ((leptons[1])+(bJets[0])).M();
+                           lep1_jet2_mass = ((leptons[1])+(Jets[Jet2_nonB])).M();
+                           lep2_jet1_mass = ((leptons[2])+(bJets[0])).M();
+                           lep2_jet2_mass = ((leptons[2])+(Jets[Jet2_nonB])).M();
+                       }
+
+                       else if(Nbjet > 1){
+                           lep1_jet1_mass = ((leptons[1])+(bJets[Jet1_B])).M();
+                           lep1_jet2_mass = ((leptons[1])+(bJets[Jet2_B])).M();
+                           lep2_jet1_mass = ((leptons[2])+(bJets[Jet1_B])).M();
+                           lep2_jet2_mass = ((leptons[2])+(bJets[Jet2_B])).M();
+                       }
+
```

Comment:
Once the jets are chosen, define the candidate jets and generalize this section to less blocks of if/else statements.